tests/test_utils.py

```diff
@@ -496,3 +497,18 @@ def test_generate_csl_given_name_with_given(self):
         given_name = 'Cause'
         csl_given_name = generate_csl_given_name(given_name)
         assert_equal(csl_given_name, 'Cause')
+
+
+@pytest.mark.django_db
+class TestUserFactoryConflict:
+
+    def test_build_create_user_time_conflict(self):
+        # Test that build and create user factories do not create conflicting usernames 
+        # because they occured quickly
+        user_one_build = UserFactory.build()
```

Comment:
I suspect that just calling fake_email twice in a row (as done in https://github.com/CenterForOpenScience/osf.io/pull/7757/files#diff-a4f5c4d64d4811b15e1405573c00f3f4R2165 ) will be a tiny bit faster than even this. Could you add a couple of lines like that with an assert just to make sure?